x-pack/plugins/ml/public/application/services/results_service/results_service.js

```diff
@@ -1232,13 +1228,13 @@ export function resultsServiceProvider(mlApiServices) {
           .esSearch({
             index,
             body,
-            rest_total_hits_as_int: true,
+            track_total_hits: true,
           })
           .then((resp) => {
             // Because of the sampling, results of metricFunctions which use sum or count
             // can be significantly skewed. Taking into account totalHits we calculate a
             // a factor to normalize results for these metricFunctions.
-            const totalHits = get(resp, ['hits', 'total'], 0);
+            const totalHits = get(resp, ['hits', 'total', 'value'], 0);
```

Comment:
It might be worth taking the opportunity to remove `get` in these cases, as we're touching this code?